### ISSUE TITLE
PLEX scrobbler bug when expanding music player

### ISSUE THREAD

**Issue by Botanistt** [web-scrobbler/web-scrobbler#1705]
Scrobbling with plex works just fine normally, but sometimes I want to expand the music player, because it would be more comfortable for me, but once I do it starts scrobbling songs incorrectly. I have attached a screenshot above where you can see the problem in action.
https://i.imgur.com/MZxZwJP.jpg
In some cases the song name would repeat multiple times, but in the screenshot you can see that it is being repeated only twice though it can be repeated even 5 times. It does scrobble it too, which obviously is not correct.

I believe this could be fixed.

The only way how to replicate this bug is if you listen on your own and enable the extended music player view. That's all.

Not sure how these are relevant to the issue, but sure.
Browser: Opera version 55.0.2994.44
Scrobbler version: 2.6.0

**Comment by inverse** [comment]
Thanks for reporting. Hopefully someone also who uses plex can reproduce and fix :+1:

**Comment by Botanistt** [comment]
anyone using PLEX? 🙄

**Comment by jaccarmac** [comment]
Is this still a problem? I see the connector was updated November 3, but if it still is I have an idea how to fix this bug given something similar I encountered in the TIDAL connector.

**Comment by jaccarmac** [comment]
Also might be worth pinging the author of the commit I'm talking about: b9e063a39530684ff2ef56aa81a35fd021875994

**Comment by alexesprit** [comment]
> I have an idea how to fix this bug given something similar I encountered in the TIDAL connector.

@jaccarmac I can reproduce this issue. What's your idea?

**Comment by jaccarmac** [comment]
My first attempt at the TIDAL connector was d817566fed098c763c84bac55c5b8a859fb3974c, but when expanding the player *there*, another `infoTable` would be created and I would get garbage data. I fixed this in b194990379c97a0bee7e87cedf2b4046526ad8f9 with `nth-child(1)`, though the fix could also be implemented in the `Connector` base class. I don't think there is any case where we would want to pull data from anything other than the first result of a given selector?

**Comment by alexesprit** [comment]
I updated `Connector.playerSelector` property and fixed this issue. Now I'm trying to fix play/pause toggle.

> though the fix could also be implemented in the Connector base class. I don't think there is any case where we would want to pull data from anything other than the first result of a given selector?

Do you mean we should replace base implementation with the following:
```js
/**
 * Default implementation of track name lookup by selector.
 *
 * Override this method for more complex behaviour.
 *
 * @return {String} Song title
 */
this.getTrack = () => $(this.trackSelector).first().text();
```

**Comment by alexesprit** [comment]
I also noticed there's a unified player for both remote and local web UI. I dropped support for legacy web UI (which was named "local UI").

**Comment by jaccarmac** [comment]
`.first()` is *probably* OK, but I'd be a little leery of it. If there's a way to grab the first non-empty selector rather than just the first one it would probably be safer.

**Comment by jaccarmac** [comment]
To clarify, it's fine for my purposes with TIDAL but I could see situations where there might be an empty element somewhere gumming us up.

**Comment by alexesprit** [comment]
Closed this in 830893e297dd5f4a40182e38ac685432a9180af8.
